ReadASCIIApp/src/ReadASCII.cpp

```diff
@@ -38,663 +38,666 @@ ReadASCII::ReadASCII(const char *portName, const char *searchDir)
-	pI_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
-	pD_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
-	pMaxHeat_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
-
-	//Init
-	setStringParam(P_Dir, "Default.txt");
-	setStringParam(P_DirBase, searchDir);
-
-	setDoubleParam(P_RampRate, 1.0);
-	setIntegerParam(P_Ramping, 0);
-	setIntegerParam(P_RampOn, 0);
-	setIntegerParam(P_LookUpOn, 0);
-
-	// Do no initialise P_SPOut here because it will be sent to the eurotherm
-
-	lastModified = 0;
-	fileBad = true; //Set so that program doesn't attempt to read file before base dir is set
-	rowNum = 0;
-
-	/* Create the thread that watches the file in the background 	*/
-	status = (asynStatus)(epicsThreadCreate("ReadASCIIFile",
-		epicsThreadPriorityMedium,
-		epicsThreadGetStackSize(epicsThreadStackMedium),
-		(EPICSTHREADFUNC)::readFilePoll,
-		this) == NULL);
-	/* Create the thread that ramps the output 	*/
-	if (status == asynSuccess)
-	{	
+    eventId_ = epicsEventCreate(epicsEventEmpty);
+
+    createParam(P_DirString, asynParamOctet, &P_Dir);
+    createParam(P_DirBaseString, asynParamOctet, &P_DirBase);
+    createParam(P_IndexString, asynParamInt32, &P_Index);
+
+    createParam(P_SPArrString, asynParamFloat64Array, &P_SPArr);
+    createParam(P_PArrString, asynParamFloat64Array, &P_PArr);
+    createParam(P_IArrString, asynParamFloat64Array, &P_IArr);
+    createParam(P_DArrString, asynParamFloat64Array, &P_DArr);
+    createParam(P_MaxHeatArrString, asynParamFloat64Array, &P_MaxHeatArr);
+
+    createParam(P_RampingString, asynParamInt32, &P_Ramping);
+    createParam(P_RampOnString, asynParamInt32, &P_RampOn);
+    createParam(P_LookUpOnString, asynParamInt32, &P_LookUpOn);
+
+    createParam(P_TargetString, asynParamFloat64, &P_Target);
+    createParam(P_SPRBVString, asynParamFloat64, &P_SPRBV);
+    createParam(P_RampRateString, asynParamFloat64, &P_RampRate);
+    createParam(P_StepsPerMinString, asynParamFloat64, &P_StepsPerMin);
+    createParam(P_CurTempString, asynParamFloat64, &P_CurTemp);
+
+    createParam(P_SPOutString, asynParamFloat64, &P_SPOut);
+    createParam(P_PString, asynParamFloat64, &P_P);
+    createParam(P_IString, asynParamFloat64, &P_I);
+    createParam(P_DString, asynParamFloat64, &P_D);
+    createParam(P_MaxHeatString, asynParamFloat64, &P_MaxHeat);
+    
+    //Allocate column data
+    pSP_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
+    pP_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
+    pI_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
+    pD_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
+    pMaxHeat_ = (epicsFloat64 *)calloc(INIT_ROW_NUM, sizeof(epicsFloat64));
+
+    //Init
+    setStringParam(P_Dir, "Default.txt");
+    setStringParam(P_DirBase, searchDir);
+
+    setDoubleParam(P_RampRate, 1.0);
+    setDoubleParam(P_StepsPerMin, 20.0);
```

Comment:
Just realized after approving this PR that this function will cause KEPCO IOC not to start with any value other than 20. Not sure if we need to set the default value here as it is passed from the DB file.

Reply:
Good spot, should now be fixed